addons/im_livechat/static/tests/new/livechat_service_tests.js

```diff
@@ -22,3 +29,21 @@ QUnit.test("persisted session history", async (assert) => {
     const { root } = await start();
     assert.containsOnce(root, ".o-mail-Message:contains(Old message in history)");
 });
+
+QUnit.test("previous operator used when available", async (assert) => {
```

Comment:
Test is not functionally meaningful enough, in terms of wording of named test, but also what it asserts.

If the current visitor had a conversation with an operator, and this operator is available, then picking up the livechat by visitor should keep same operator.

- The `get_session` assertion is too technical, not worth having it.
- The assert of params `previous_operator_id` is also too technical, and doesn't ensure enough the livechat operator is really the same (can easily be deadcode in args)

Would be much better to let visitor open livechat, and then see visual clue that operator is the same (or if there's none, reading python model or making `channel_info` is more trustworthy than this `mockRPC`)

Reply:
Task to improve this test added :)